cmd/api/src/api/v2/saved_queries.go

```diff
@@ -100,3 +102,32 @@ func (s Resources) CreateSavedQuery(response http.ResponseWriter, request *http.
 		api.WriteBasicResponse(request.Context(), savedQuery, http.StatusCreated, response)
 	}
 }
+
+func (s Resources) DeleteSavedQuery(response http.ResponseWriter, request *http.Request) {
+	var (
+		rawSavedQueryID = mux.Vars(request)[api.URIPathVariableSavedQueryID]
+		userID          = ctx2.FromRequest(request).AuthCtx.Session.UserID
+	)
+
+	if savedQueryID, err := strconv.Atoi(rawSavedQueryID); err != nil {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, api.ErrorResponseDetailsIDMalformed, request), response)
+	} else if savedQueries, _, err := s.DB.ListSavedQueries(userID, "", model.SQLFilter{}, 0, 10000); err != nil {
+		api.HandleDatabaseError(request, response, err)
+	} else {
+		// ensure that the query ID specified belongs to the current user
+		savedQueryIDMatched := false
+		for _, savedQuery := range savedQueries {
+			if savedQuery.ID == int64(savedQueryID) {
+				savedQueryIDMatched = true
+				break
+			}
+		}
+
+		if !savedQueryIDMatched {
+			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, "invalid saved_query_id supplied", request), response)
+		} else {
+			s.DB.DeleteSavedQuery(savedQueryID)
+			response.WriteHeader(http.StatusNoContent)
```

Comment:
This is ineffective, as you can see that a successful delete gives a 200 response currently, likely due to a middleware seeing that we haven't sent a response and somehow overriding our status header